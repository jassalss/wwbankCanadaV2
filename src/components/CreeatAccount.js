import React, { Component } from "react";
import { saveNewCustomer } from "../actions";
import { connect } from "react-redux";
import Converter from "../Converter";
class CreeatAccount extends Component {
  handleSubmit = (event) => {
    event.preventDefault();
    let CustomerName = event.target.customerName.value;
    let CustomerId = event.target.customerId.value;
    let AccountNumber = event.target.accountNumber.value;
    let Balance = event.target.balance.value;
    let currency = event.target.currency.value;
    if (Balance === "") {
      Balance = 0;
    }
    let otherCurrency = "";
    if (currency !== "CAD") {
      otherCurrency = ` = $${Balance} ${currency}`;
      Balance = Converter(Balance, currency);
    }
    let objectToBeSaved = {
      CustomerName,
      CustomerId,
      AccountNumber,
      Balance,
    };
    this.props.saveNewCustomer(objectToBeSaved, otherCurrency);
  };
  render() {
    if (this.props.submitedMessage) {
      alert(this.props.submitedMessage);
      window.location.reload();
    }
    return (
      <div>
        <h3 className="ui center aligned header">Create Customer</h3>
        <h4 className="ui center aligned header">Admin Only</h4>
        <div className="ui centered grid">
          <div className="ten wide column">
            <form className="ui form" onSubmit={this.handleSubmit}>
              <div className="field">
                <label>Customer Name</label>
                <input
                  required
                  type="text"
                  name="customerName"
                  placeholder="Customer Name"
                />
              </div>
              <div className="field">
                <label>Customer ID</label>
                <input
                  required
                  type="number"
                  name="customerId"
                  placeholder="Customer ID (Input is Number not letter)"
                />
              </div>
              <div className="field">
                <label>Account Number</label>
                <input
                  required
                  type="number"
                  name="accountNumber"
                  placeholder="Account Number (Input is Number not letter)"
                />
              </div>
              <div className="field">
                <select class="ui dropdown" required name="currency">
                  <option value="">Select Currency</option>
                  <option value="CAD">CAD</option>
                  <option value="USD">USD</option>
                  <option value="MXN">MXN</option>
                </select>
              </div>
              <div className="field">
                <label>Initial Balance</label>
                <input
                  type="number"
                  name="balance"
                  step="0.001"
                  placeholder="Initial Balance (Input is Number not letter)"
                />
              </div>

              <button className="ui button" type="submit">
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    submitedMessage: state.bankAccountsInfo.savingMessage,
  };
};
export default connect(mapStateToProps, { saveNewCustomer })(CreeatAccount);
