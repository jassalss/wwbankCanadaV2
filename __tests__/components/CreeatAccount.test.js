import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import "babel-polyfill";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
const mockStore = configureStore([]);
let bankAccountsInfo = {
  savingMessage: "test",
  depositInfo: "test",
  withDrawInfo: "test",
  trasferInfo: "test",
};
import CreeatAccount from "../../src/components/CreeatAccount";
window.alert = jest.fn();
describe("Testing create account page with Snapshot", () => {
  let store;
  beforeEach(() => {
    store = mockStore({
      bankAccountsInfo,
    });
  });

  it("matches the snapshot", () => {
    const { location } = window;
    delete window.location;
    window.location = { reload: jest.fn() };
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CreeatAccount />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    window.location = location;
  });
  window.alert.mockClear();
});
