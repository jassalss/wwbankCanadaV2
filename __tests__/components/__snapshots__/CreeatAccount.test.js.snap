// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing create account page with Snapshot matches the snapshot 1`] = `
<div>
  <h3
    className="ui center aligned header"
  >
    Create Customer
  </h3>
  <h4
    className="ui center aligned header"
  >
    Admin Only
  </h4>
  <div
    className="ui centered grid"
  >
    <div
      className="ten wide column"
    >
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="field"
        >
          <label>
            Customer Name
          </label>
          <input
            name="customerName"
            placeholder="Customer Name"
            required={true}
            type="text"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Customer ID
          </label>
          <input
            name="customerId"
            placeholder="Customer ID (Input is Number not letter)"
            required={true}
            type="number"
          />
        </div>
        <div
          className="field"
        >
          <label>
            Account Number
          </label>
          <input
            name="accountNumber"
            placeholder="Account Number (Input is Number not letter)"
            required={true}
            type="number"
          />
        </div>
        <div
          className="field"
        >
          <select
            class="ui dropdown"
            name="currency"
            required={true}
          >
            <option
              value=""
            >
              Select Currency
            </option>
            <option
              value="CAD"
            >
              CAD
            </option>
            <option
              value="USD"
            >
              USD
            </option>
            <option
              value="MXN"
            >
              MXN
            </option>
          </select>
        </div>
        <div
          className="field"
        >
          <label>
            Initial Balance
          </label>
          <input
            name="balance"
            placeholder="Initial Balance (Input is Number not letter)"
            step="0.001"
            type="number"
          />
        </div>
        <button
          className="ui button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
